src/zod.ts

```diff
@@ -0,0 +1,211 @@
+import { z } from "zod";
+import { Action, LinkType, MatchMode } from "./constants.js";
+import { existsSync } from "fs";
+import { logger } from "./logger.js";
+import { getFileConfig } from "./configuration.js";
+
+const fileConfig = await getFileConfig();
+
+/**
+ * VALIDATION_SCHEMA is just a object of all the zod schemas
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		notificationWebhookUrl: z.union([
+			z.undefined(),
+			z
+				.string()
+				.url()
+				.refine((url) => ({
+					message: `invalid notificationWebhookUrl URL - ${url}`,
+				})),
+		]),
+		torznab: z.array(
+			z
+				.string()
+				.url()
+				.refine((url) => ({
+					message: `invalid torznab URL - ${url}`,
+				}))
+		),
+		dataDirs: z.union([
+			z.undefined(),
+			z.array(
+				z.string().refine(
+					(path) => existsSync(path),
+					(path) => ({
+						message: `the dataDirs path ${path} does not exist on the filesystem`,
+					})
+				)
+			),
+		]),
+		torrentDir: z.string().refine(
+			(path) => existsSync(path),
+			(path) => ({
+				message: `the torrentDir path ${path} does not exist on the filesystem`,
+			})
+		),
+		linkDir: z.union([
+			z.undefined(),
+			z.string().refine(
+				(path) => existsSync(path),
+				(path) => ({
+					message: `the linkDir path ${path} does not exist on the filesystem`,
+				})
+			),
+		]),
+		outputDir: z.string().refine(
+			(path) => existsSync(path),
+			(path) => ({
+				message: `the outputDir path ${path} does not exist on the filesystem`,
+			})
+		),
+		includeEpisodes: z.boolean({
+			invalid_type_error:
+				"includeEpisodes must be a defined boolean (true or false)",
+		}),
+		includeSingleEpisodes: z.boolean({
+			invalid_type_error:
+				"includeSingleEpisodes must be a defined boolean (true or false)",
+		}),
+		includeNonVideos: z.boolean({
+			invalid_type_error:
+				"includeNonVideos must be a defined boolean (true or false)",
+		}),
+		duplicateCategories: z.boolean({
+			invalid_type_error:
```

Comment:
```
info: cross-seed v5.5.4
info: Validating your configuration...
error: [
  {
    "code": "invalid_type",
    "expected": "boolean",
    "received": "undefined",
    "path": [
      "skipRecheck"
    ],
    "message": "Required"
  }
]
```

This was what happened when I ran it without, skipRecheck set to "false" (string) instead of `false` (boolean)

Reply:
This is what displays when you have a message

```
{
    "code": "invalid_type",
    "expected": "number",
    "received": "string",
    "path": [
      "searchLimit"
    ],
    "message": "searchLimit must be a number or undefined"
  }
```

Reply:
I wonder if we could read the error object and assemble the human friendly error message from that instead of maintaining all of these repetitive strings

Reply:
I tried this, and it wasn't an actual array, it was type ZodError, I just said this in discord lol

If you can help me figure out how to typecast it into something I can parse as an array, that would be ideal, but other than by \n strip, I'm not sure how to do this honestly.

Reply:
https://deno.land/x/zod@v3-snapshot-2021-01-21/ERROR_HANDLING.md?source=#error-handling-in-zod